src/test/java/org/matsim/prepare/longDistanceFreightGER/tripGeneration/GenerateFreightTripTest.java

```diff
@@ -0,0 +1,72 @@
+package org.matsim.prepare.longDistanceFreightGER.tripGeneration;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.matsim.api.core.v01.network.Network;
+import org.matsim.api.core.v01.population.Person;
+import org.matsim.api.core.v01.population.Population;
+import org.matsim.api.core.v01.population.PopulationWriter;
+import org.matsim.core.config.ConfigUtils;
+import org.matsim.core.network.NetworkUtils;
+import org.matsim.core.population.PopulationUtils;
+import org.matsim.testcases.MatsimTestUtils;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.matsim.prepare.longDistanceFreightGER.tripGeneration.GenerateFreightPlans.createOutput_tripOD_relations;
+
+public class GenerateFreightTripTest {
+
+	private static final Logger log = LogManager.getLogger(GenerateFreightTripTest.class);
+
+	@RegisterExtension
+	public MatsimTestUtils utils = new MatsimTestUtils() ;
+
+	// TODO the test fails... But the main class with the same input arguments can run without problem... I don't know why
+	@Test
+	public final void test() {
+		String [] args = {
+			"--output", utils.getOutputDirectory(),
+			"--sample", "0.001",
+			"--land-use-filter"
+		};
+
+		GenerateFreightPlans.main(args);
+	}
+
+	@Test
+	public final void test2() throws IOException {
+		Network network = NetworkUtils.readNetwork("https://svn.vsp.tu-berlin.de/repos/public-svn/matsim/scenarios/countries/de/german-wide-freight/v2/germany-europe-network.xml.gz");
+		FreightAgentGenerator freightAgentGenerator = new FreightAgentGenerator(network,
+			"https://svn.vsp.tu-berlin.de/repos/public-svn/matsim/scenarios/countries/de/german-wide-freight/raw-data/shp/NUTS_RG_20M_2016_4326.shp/NUTS_RG_20M_2016_4326.shp",
+			null, 13, 260, 0.001 / 100);
+		log.info("Freight agent generator successfully created!");
+
+		log.info("Reading trip relations...");
+		List<TripRelation> tripRelations = TripRelation.readTripRelations("https://svn.vsp.tu-berlin.de/repos/public-svn/matsim/scenarios/countries/de/german-wide-freight/raw-data/ketten-2010.csv");
+		log.info("Trip relations successfully loaded. There are {} trip relations", tripRelations.size());
+
+		log.info("Start generating population...");
+		Population outputPopulation = PopulationUtils.createPopulation(ConfigUtils.createConfig());
+		for (int i = 0; i < tripRelations.size(); i++) {
+			List<Person> persons = freightAgentGenerator.generateFreightAgents(tripRelations.get(i), Integer.toString(i));
+			for (Person person : persons) {
+				outputPopulation.addPerson(person);
+			}
+		}
```

Comment:
Same here: You can at least check the number of persons.
And even better, check additionally all attributes for 1 specific person.